### PR TITLE
use subtest for table units (pkg/kubectl)

### PR DESCRIPTION
**Pull request by wgliang**
**What this PR does / why we need it**:
Go 1.7 added the subtest feature which can make table-driven tests much easier to run and debug. Many table-driven tests in pkg/kubectl are not using this feature.

/kind cleanup
/area kubectl

Further reading: [Using Subtests and Sub-benchmarks](https://blog.golang.org/subtests)

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by apelisse** [comment]
It's funny, I just discovered this feature today while reading the testing godoc :-)

I'll try to have a look. I think you should have split the PR into smaller pieces because now I'll feel bad leaving comments, forcing you to change a gazillion files ...

**Review by apelisse — COMMENTED** [review]
While I think this could be improved, this is already a big step in the right direction, so you'll have my lgtm after rebase.

We're in a deep need for more and better unit-tests, so this work is really appreciated! thank you.

File: pkg/kubectl/cmd/config/create_authinfo_test.go [hunk 1]
```diff
@@ -141,55 +150,58 @@ func TestCreateAuthInfoOptions(t *testing.T) {
...
 			flags: []string{
 				// No name for authinfo provided.
 			},
 			wantCompleteErr: true,
 		},
 	}
 
-	for i, test := range tests {
-		buff := new(bytes.Buffer)
+	for i, tt := range tests {
```
[the remark this answers is not in the document]
Comment (apelisse) [reply pkg/kubectl/cmd/config/create_authinfo_test.go]: Actually, you could just remove the "testX" everywhere, and generate that name from `i`: `fmt.Sprintf("test%d", i)`

**Comment by apelisse** [comment]
Clearly it's broken.

**Comment by apelisse** [comment]
OK, Looks like the "pr needs rebase" label is missing

File: pkg/kubectl/pdb_test.go [hunk 2]
```diff
@@ -36,12 +36,14 @@ func TestPodDisruptionBudgetV1Generate(t *testing.T) {
 		t.Errorf("unexpected error: %v", err)
 	}
 
-	tests := map[string]struct {
+	tests := []struct {
```
Comment (apelisse) [inline pkg/kubectl/pdb_test.go]: that change isn't really needed

File: pkg/kubectl/pdb_test.go [hunk 3]
```diff
@@ -170,12 +183,14 @@ func TestPodDisruptionBudgetV2Generate(t *testing.T) {
...
```
Comment (apelisse) [inline pkg/kubectl/pdb_test.go]: I like maps as much as I like arrays

Comment (wgliang) [reply pkg/kubectl/pdb_test.go]: I can't agree that, no matter what else, the use of arrays in kubernetes is a bit more.

**Comment by wgliang** [comment]
/ping @apelisse @mengqiy 
PTAL or what I should do.

**Comment by apelisse** [comment]
Sorry for the delay :-)

**Pull request merged**